sharpy/generators/dynamiccontrolsurface.py

```diff
@@ -2,66 +2,85 @@
 
 import sharpy.utils.generator_interface as generator_interface
 import sharpy.utils.settings as settings
+import sharpy.utils.cout_utils as cout_utils
 
 
 @generator_interface.generator
 class DynamicControlSurface(generator_interface.BaseGenerator):
     """
     Dynamic Control Surface deflection Generator
 
-    ``DynamicControlSurface`` class inherited from ``BaseGenerator``
+    The object generates a deflection in radians based on the time series given as a single vector in the input data.
+    A first order finite-differences scheme is used to calculate the deflection rate based on the provided time step
+    increment.
 
-    The object generates a deflection in radians based on the time series given as a vector in the input data
+    To call this generator, the ``generator_id = DynamicControlSurface`` key shall be used for the setting
+    `control_surface_deflection` in the ``AerogridLoader`` solver.
 
-    To call this generator, the ``generator_id = DynamicControlSurface`` shall be used.
-    This is parsed as the value for the ``control_surface_deflection_generator`` key in the aerogridloader solver's settings.
+    These generator settings should be parsed as a dictionary to the setting
+    ``control_surface_deflection_generator_settings`` in ``AerogridLoader``, where this dictionary should be the value
+    to another dictionary where the key is the index of the control surface in string format. This is shown better
+    in the example below:
+
+    Examples:
+
+        .. code-block:
 
-    Args:
-        in_dict (dict): Input data in the form of dictionary. See acceptable entries below.
+            cs0_settings = {}  # these are the settings for the desired control surface
+            settings = {}
+            settings['AerogridLoader] = {'control_surface_deflection' : ['DynamicControlSurface],
+                                         'control_surface_deflection_generator_settings: {'0': cs_settings},
+                                         }
```

Comment:
```suggestion
            cs0_settings = {}  # these are the settings for control surface number 0
            cs1_settings = {}  # these are the settings for control surface number 1
            dict_of_cs = {'0': cs0_settings,
                                  '1': cs1_settings} # This dictionary groups all the settings for all the control surfaces
            settings = {}
            settings['AerogridLoader] = {'control_surface_deflection' : ['DynamicControlSurface],
                                                         'control_surface_deflection_generator_settings: dict_of_cs}
```

Same as the above comment